src/Avalonia.Controls/AppBuilderBase.cs

```diff
@@ -126,8 +126,7 @@ public void Start<TMainWindow>(Func<object> dataContextProvider = null)
             
             // Copy-pasted because we can't call extension methods due to generic constraints
             var lifetime = new ClassicDesktopStyleApplicationLifetime(Instance) {ShutdownMode = ShutdownMode.OnMainWindowClose};
```

Comment:
btw here `Instance` is null and `ClassicDesktopStyleApplicationLifetime` in general is always having a null application.
it's good that  `app.Run(...)` is extension method and everything is working fine, but there might be some general architecture design vulnerabilities

Reply:
That extension method is a compatibility one. The lifetime itself doesn't need the _app at all.